internal/provider/resource_gitlab_runner.go

```diff
@@ -0,0 +1,257 @@
+	// lintignore: XR001 // TODO: replace with alternative for GetOkExists
+	if v, ok := d.GetOkExists("locked"); ok {
+		options.Locked = gitlab.Bool(v.(bool))
+	}
+
+	// nolint:staticcheck // SA1019 ignore deprecated GetOkExists
+	// lintignore: XR001 // TODO: replace with alternative for GetOkExists
+	if v, ok := d.GetOkExists("run_untagged"); ok {
+		options.RunUntagged = gitlab.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("tag_list"); ok {
+		options.TagList = stringListToStringSlice(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("access_level"); ok {
+		options.AccessLevel = gitlab.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("maximum_timeout"); ok {
+		options.MaximumTimeout = gitlab.Int(v.(int))
+	}
+
+	log.Printf("[DEBUG] Update GitLab Runner %s", d.Id())
+	_, _, err := client.Runners.UpdateRunnerDetails(runnerId, options, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceGitLabRunnerRead(ctx, d, meta)
+
+}
+
+func resourceGitLabRunnerCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	options := &gitlab.RegisterNewRunnerOptions{
+		Token: gitlab.String(d.Get("token").(string)),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		options.Description = gitlab.String(v.(string))
+	}
+
+	//GetOK skips the block if the value is "false", so need to use GetOkExists even though it's deprecated.
+	// nolint:staticcheck // SA1019 ignore deprecated GetOkExists
+	// lintignore: XR001 // TODO: replace with alternative for GetOkExists
+	if v, ok := d.GetOkExists("paused"); ok {
+		options.Paused = gitlab.Bool(v.(bool))
+	}
+
+	// nolint:staticcheck // SA1019 ignore deprecated GetOkExist
+	// lintignore: XR001 // TODO: replace with alternative for GetOkExists
+	if v, ok := d.GetOkExists("locked"); ok {
+		options.Locked = gitlab.Bool(v.(bool))
+	}
+
+	// nolint:staticcheck // SA1019 ignore deprecated GetOkExists
+	// lintignore: XR001 // TODO: replace with alternative for GetOkExists
+	if v, ok := d.GetOkExists("run_untagged"); ok {
+		options.RunUntagged = gitlab.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("tag_list"); ok {
+		options.TagList = stringListToStringSlice(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("access_level"); ok {
+		options.AccessLevel = gitlab.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("maximum_timeout"); ok {
+		options.MaximumTimeout = gitlab.Int(v.(int))
+	}
+
+	// Explicitly not printing the registration token here, even though it may make debugging a bit trickier, since it's a
+	// secret.
+	log.Printf("[DEBUG] Update GitLab Runner using registration token in configuration")
+	runner, _, err := client.Runners.RegisterNewRunner(options, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(strconv.Itoa(runner.ID))
+
+	//The authentication_token will ONLY exist during creation, and will not return during "read", so we need to set it here.
```

Comment:
NIT: in general I think it's nice to have a space between `//` and the first character .. just makes it more readable ... but major personal opinion, so 🤷  😅 

```suggestion
	// The authentication_token will ONLY exist during creation, and will not return during "read", so we need to set it here.
```